src/components/Button/PeriodToggleButton.jsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from "react";
+
+import { PERIOD_TYPE } from "../../config/constants";
+import PropTypes from "prop-types";
+
+const PeriodToggleButton = ({ period, setPeriod, setCursorId }) => {
+  const [toggleOnPeriod, setToggleOnPeriod] = useState(() => period);
```

Comment:
```suggestion
  const [toggleOnPeriod, setToggleOnPeriod] = useState(period);
```
`toggleOnPeriod`의 초기값으로 `() => period`를 콜백 함수로 설정하신 맥락이 궁금합니다!

콜백 함수에서도 내부 동작 없이 바로 `period`를 return 하기 때문에 위와 같이 수정해볼 수  있다고 생각했어요. 초기에 한 번만 설정되는 동작도 동일하고요. 혹시 다른 의도가 있으셨을지 여쭙습니다.

참고 [React - Avoiding recreating the initial state](https://react.dev/reference/react/useState#avoiding-recreating-the-initial-state)

Reply:
혹시 `toggleOnPeriod`라는 새로운 상태를 만드신 이유가 무엇인지 궁금합니다. 현재 `period`와 `setPeriod`라는 상태와 setter 함수를 내리고 있는데요, 그 상태와 함수를 바로 활용하는 것이 어떨지 제안드립니다. 왜냐하면, period라는 부모에서 선언한 상태를 변형하여 사용하는 것도 아니기 때문에, 그대로 쓰면 될 것 같아요~

cc @jin-ttao

Reply:
@jin-ttao 
초기값으로 `() => period`로 설정한 이유는 해당 컴포넌트가 맨 처음 실행될 때, props인 `period`가 아직 전달되지 않아 
undefined로 설정될 수 있기 때문에 lazy initialization 방법을 통해서 온전한 period 값으로 설정될 수 있게 하기 위해서였습니다. 

이전에 과제할 때 props로 전달받은 데이터를 state의 초기값으로 설정해야 했을 때, `undefined`로 초기값이 설정되어 
해당 문제의 원인을 찾다가 알게된 방식인데요. 아래 규경님의 코멘트와 같이 굳이 state로 선언하지 않아도 되었네요!ㅎㅎ 

제가 참고했던 [lazy initialization에 대한 개념을 정리한 블로그](https://stay-present.tistory.com/128) 링크 걸어두겠습니다!